### ISSUE TITLE
Issue with CGU validation

### ISSUE THREAD

**Issue by johnkrovitch** [dis-moi/extension#634]
Hi,

I have a strange bug with CGU validation : I need to validate the CGU multiple times. It is like I need to validate for each site managed by the extension.

For example, I browse amazon.fr, the extension asks me to validate the CGU. I click and do validate. Then I go to another site, airbnb.fr, then the extension asks again me to validate the CGU.

But the strange thing is that the bug does not appear every time. Sometime I validate the CGU once, and it works.

It is like the synchronization is not working everytime.

I'm using Firefox on Windows and Linux (KDE Neon), and use Firefox synchronization (history...) between each instance of Firefox.

Thanks

**Comment by MaartenLMEM** [comment]
Thanks a lot John !
There is obviously a problem here. With a colleague, we tried to reproduce the bug, we didn't success.
But last days a dozen of people who uninstalled Dismoi told "it requested a new permission". 
Also one of us had same problem as you but just one time.
We are trying to understand and find the origin of it. 
If you agree we will contact you for some more details.
Best,

**Comment by johnkrovitch** [comment]
@MaartenLMEM yes of course.

**Comment by lutangar** [comment]
Thank you for reporting this issue! 
We'll need a few more informations before digging further :
- Approximate date of first installation
- Current Dismoi extension version number (check about:addons in Firefox)

I'll get back to you asap.

**Comment by johnkrovitch** [comment]
Hi,
the extension version is 3.30.0 (and Firefox 76.0.1 (64 bits)).
The first installation was a long time ago, when it was called "le même en mieux". I do not know the precise date.

**Comment by johnkrovitch** [comment]
Hello,

I may have new data on the bug : I had the issue in my Linux browser (Firefox 76.0.1). It seems that the extension tells me to validate the CGU only if there is a contribution on the current page. Otherwise, it seems that have do not have the issue.

**Comment by lutangar** [comment]
After digging on lost settings it's seems to be a recurring issue with extensions in general. 

See the following :
- https://help.getadblock.com/support/solutions/articles/6000169060-i-keep-seeing-your-payment-page-or-adblock-keeps-losing-my-settings-in-firefox
- https://www.reddit.com/r/firefox/comments/bwv68f/how_to_stop_firefox_from_automatically_resetting/

We decided to change the design of the extension to fit this behavior which obviously we can't fully control.
We now consider that the ToS (CGU) have been accepted when the extension is installed.
> See more details in #666

We also made adjustments to make the extension painless to use without a state.
